tt_metal/jit_build/build.cpp

```diff
@@ -558,6 +559,18 @@ void JitBuildState::weaken(const string& log_file, const string& out_dir) const
     }
 }
 
+void JitBuildState::extract_zone_src_locations(const string& log_file) const
+{
+    static std::atomic<bool> new_log = true;
+    if (tt::tt_metal::getDeviceProfilerState()) {
+        if (new_log.exchange(false) && std::filesystem::exists(tt::tt_metal::PROFILER_ZONE_SRC_LOCATIONS_LOG)) {
```

Comment:
can this race against other processes (depends where PROFILER_ZONE_SRC_LOCATIONS exists, device specific?). 

use lock_guard?

Reply:
I believe because  it is and with what is in `new_log` and `new_log` is atomic and the very first atomic exchange (fetch and set to false atomically) sets it to false all other concurrent calls will see false for the expression. 

Is it cleaner to bring the file exists check inside the scope of the flag check if? Avoid && all together?

Reply:
oh, right, I get it.  fine as is